src/poetry/console/commands/run.py

```diff
@@ -44,7 +44,13 @@ def _module(self) -> Module:
 
         return module
 
-    def run_script(self, script: str | dict[str, str], args: str) -> int:
+    def run_script(self, script: str | dict[str, str], args: list[str]) -> int:
+        # Calling `sys.argv` should run the same program as the currently
```

Comment:
Let's redo this comment as a docstring, e.g. 

```python
"""
This function runs the script, setting argv[0] because...
"""
```

Reply:
I redid as the following:

```python
"""Runs an entry point script defined in the section ``[tool.poetry.scripts]``.

When a script exists in the venv bin folder, i.e. after ``poetry install``,
then ``sys.argv[0]`` must be set to the full path of the executable, so
``poetry run foo`` and ``poetry shell``, ``foo`` have the same ``sys.argv[0]``
that points to the full path.

Otherwise (when an entry point script does not exist), ``sys.argv[0]`` is the
script name only, i.e. ``poetry run foo`` has ``sys.argv == ['foo']``.
"""
```